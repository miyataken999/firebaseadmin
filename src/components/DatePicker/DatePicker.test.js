import React from 'react';

import DatePickerStyled from '.';

describe('<DatePickerStyled /> rendering', () => {
  const onChange = jest.fn();

  it('should render without crashing', () => {
    const { component } = renderWithProviders(
      <DatePickerStyled
        name="test"
        dateFormat="en-US"
        date={new Date('11/12/2020')}
        onChange={onChange}
      />
    )({});

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render <DatePicker /> component correctly', () => {
    const { component } = renderWithProviders(
      <DatePickerStyled
        name="test"
        dateFormat="en-US"
        date={new Date('11/12/2020')}
        onChange={onChange}
      />
    )({});

    expect(component.container.querySelector('input')).toBeTruthy();
  });

  it('should pass the date prop to <DatePicker selected={date} /> correctly', () => {
    const { component } = renderWithProviders(
      <DatePickerStyled
        name="test"
        dateFormat="en-US"
        date={new Date('11/12/2020')}
        onChange={onChange}
      />
    )({});

    expect(component.container.querySelector('input').value).toBe('11-12-20');
  });
});
