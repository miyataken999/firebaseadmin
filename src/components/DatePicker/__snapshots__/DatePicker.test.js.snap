// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePickerStyled /> rendering should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        class=""
        type="text"
        value="11-12-20"
      />
    </div>
  </div>
</DocumentFragment>
`;
